gui/drawer.go

```diff
@@ -125,17 +126,22 @@ L:
 	d.chartDrawing.Store(false)
 }
 
-func (d *drawer) redrawGauge(ctx context.Context, maxSize int) {
-	var count float64
-	size := float64(maxSize)
+func (d *drawer) redrawGauge(ctx context.Context, duration time.Duration) {
+	totalTime := float64(duration)
 	d.widgets.progressGauge.Percent(0)
-	for {
+	for start := time.Now(); ; {
 		select {
 		case <-ctx.Done():
 			return
 		case <-d.gaugeCh:
-			count++
-			d.widgets.progressGauge.Percent(int(count / size * 100))
+			passed := float64(time.Since(start))
+			percent := int(passed / totalTime * 100)
+			// as time.Duration is the unit of nanoseconds
+			// small duration can exceed 100 on slow machines
+			if percent > 100 {
+				percent = 100
```

Comment:
We'd better avoid calling Termdash API more than needed.

```suggestion
				percent = 100
				continue
```